curbmap-ios-mvp/Services/LocationServices.swift

```diff
@@ -2,14 +2,66 @@
 //  LocationServices.swift
 //  curbmap-ios-mvp
 //
-//  Created by Eli Selkin on 4/10/18.
-//  Copyright © 2018 Eli Selkin. All rights reserved.
+//  Copyright © 2018 curbmap. All rights reserved.
 //
 
 import Foundation
 import MapKit
 
-class LocationServices: NSObject {
-    public static var currentLocation = LocationServices()
+class LocationServices: NSObject, CLLocationManagerDelegate {
+    public static var currentLocation = LocationServices(true)
+    private var heading: CLHeading!
+    private var lastPosition: CLLocation!
+    private var locationManager: CLLocationManager!
+    private var following: Bool
     
+    private init(_ following: Bool) {
+        self.following = following
+        self.locationManager = CLLocationManager()
+        self.locationManager.requestWhenInUseAuthorization()
+        self.locationManager.desiredAccuracy = kCLLocationAccuracyBest
+        self.locationManager.startUpdatingLocation()
+        self.locationManager.startUpdatingHeading()
+        super.init()
+        self.locationManager.delegate = self
+    }
+    
+    public func getLocation() -> CLLocation? {
+        return locationManager.location
+    }
+    
+    public func getHeading() -> CLHeading? {
+        return locationManager.heading
+    }
+    
+    public func setFollowing(follow: Bool) -> Void {
```

Comment:
When do we set following?

Reply:
Set following should be a choice for the user. By default it should probably be set to true, until we develop settings.